### ISSUE TITLE
Multiple languages and Spell check

### ISSUE THREAD

**Issue by PierreMarchand20** [atom/spell-check#230]
I have recently upgrade atom to 1.23.1 and since then, spell checking in my latex files does not make any sense.

I have set the grammar `text.tex.latex` and the local `en-US, fr-Fr` in the settings of the spell check package and it used to work nicely. But now, it does not recognize English and French words. Even weirder, it suggests me French and English words, which means that it actually finds the right dictionaries ! For example, I write "problems" and it suggests me "problème" (a French word) and "problem" (an English word).

<img width="301" alt="capture d ecran 2017-12-20 a 10 44 46" src="https://user-images.githubusercontent.com/15056323/34200875-0087e514-e573-11e7-8023-818b1ae41696.png">

EDIT: it seems that this behavior appears if they are multiple languages defined in the setting of spell-check, but does not depend on the grammar (I tried with text.plain for instance). Thus, I changed the title of the issue.

**Comment by mbataillou** [comment]
+1

**Comment by alexdevero** [comment]
I have the same issue with locals set to `en-US, en-GB, cs-CZ`. Now, neither English nor Czech words are recognized and all of them are highlighted. However, as @PierreMarchand20 mentioned, suggestions work.

**Comment by dmoonfire** [comment]
Since I'm not sure about your environment, do you get the same thing with a Markdown or text document?

**Comment by alexdevero** [comment]
@dmoonfire text document as well (.txt):
![text](https://user-images.githubusercontent.com/4237724/36681782-dd3f1bd6-1b19-11e8-8a66-8d247a3ec067.PNG)

My OS is Window 10, if that's what you meant by environment.

**Comment by anthonykevins** [comment]
+1. Exact same issue here, though on a Mac (High Sierra). Spell check does seem to work fine in Markdown documents, however.

**Comment by zeis** [comment]
Similar problem with locales set to `it-IT, en-GB`, for some reason some words are misspelled but not others.
Atom version: 1.30.0, OSX High Sierra.

![image](https://user-images.githubusercontent.com/1945427/44958098-63c92e80-aedb-11e8-9dbf-b08169113e78.png)

**Comment by reskes** [comment]
TL:DR Check if the needed Locales are installed on your OS first. Install them and you're good to go.

I had the same issue on MacOS 10.14.2 (Mojave) with Atom 1.33.1 and spell-check 0.74.2 in Combination of german and english Locale. 
(Especially with Latex)

<img width="228" alt="bildschirmfoto 2019-01-04 um 00 12 02" src="https://user-images.githubusercontent.com/6602847/50666682-245c3d00-0fb6-11e9-8345-01ff0efb9dad.png">

No error-message or similar thing was given. I checked my Macs locale and language settings (global settings of OS).
And voila the English package was missing. Don't know why because I'm sure it was installed sometime ago.

So I installed it (again):
<img width="228" alt="bildschirmfoto 2019-01-04 um 00 12 32" src="https://user-images.githubusercontent.com/6602847/50666683-258d6a00-0fb6-11e9-9e3a-a9b716df6bac.png">

**Comment by Andrade** [comment]
In Ubuntu 18.04, the list of locales uses underscores and not dash. For example, for american english and german this would be: `en_US, de_DE`.

I know it's not macOS, but thought I'd leave it here in case it's useful for someone else.

**Comment by GGBoCa** [comment]
Hello !

One year later I seem to have the same problem. French and english (US ans UK) are installed on my os (mac). In Atom's Spell Check > Locales I wrote `fr-FR, en_US`. And it only takes one of those. French in this case and english if I remove `fr-FR`. Like you I'd like to have both.

Did you find a solution since your last message ?

Thanks a lot.

GG

**Comment by reskes** [comment]
As @Andrade the syntax for locales seem to differ from OS to OS. Try either using `fr_FR, en_US` or `fr-FR, en-US`.

**Comment by GGBoCa** [comment]
OK it works ! But it's still weird. I tried a lot possibilities. fr-FR, fr-CH, fr_fr, with comma, without comma, spaces, no space, etc.

Finally, in my case, my OS is in French and english US is install too. Then I put `en_US, fr_ch, ` with a "space" after the last comma. Like : `en_US, fr_ch,(space)`.

So I don't understand, but I don't touch anymore. Hope it can help some one else.

Tschüss
GG

**Comment by dmoonfire** [comment]
Well, that is definitely a bug with the comma-separated list. Let me see about getting a PR in to fix that. I was letting everything settle with the latest merge because it was so drastic. I also tried to set it up so it tries underscores and dashes (regardless of what you put in), but there might be a flaw there too. Thank you.
